pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -251,27 +255,36 @@ func (c *Controller) updateOperatorStatus(workload *appsv1.Deployment, operatorC
 		deploymentAvailableCondition.Reason = "AsExpected"
 	}
 
+	desiredReplicas := int32(1)
+	if workload.Spec.Replicas != nil {
+		desiredReplicas = *(workload.Spec.Replicas)
+	}
+
 	// If the workload is up to date, then we are no longer progressing
 	workloadAtHighestGeneration := workload.ObjectMeta.Generation == workload.Status.ObservedGeneration
+	workloadIsBeingUpdated := workload.Status.UpdatedReplicas < desiredReplicas
```

Comment:
this really only means that we deploy, not that we update.

Reply:
does `status.UpdatedReplicas` stay down until pods are ready?

Reply:
Nvmd, `UpdatedReplicas`, not status. replicas.

Reply:
It's non-terminated pods, so once a new pod appears, this number gets increased.

I was thinking of extending this, but that might make this PR more complicated. For now, I'm ok with what it does, but there's certainly room for improvement.